gitpod-io/gitpod
# Fix context missing for test cases

## Description


**There're some places will print `mismatch`**
- `authorizaer`: it's testing mismatch cases
- `organization-service`: testing PERMISSION_DENIED / NOT_FOUND (i.e. member update settings, stranger find org)

**Important logic change:**
- `organization-service`: `addOrUpdateMember` with a new owner it will delete BUILT_IN one from org, add a `runWithSubjectId` to make sure it's the new owner going to delete the member.
- `workspace-starter`: `failInstanceStart` it can happend no instance (at least in test case)

**Others**
- `bearer-authxxx.spec.ts`: logic of expectError is not complete
- Test cases to create orgniaztion via BUILT IN users, first joiner should be owner

## Related Issue(s)

Fixed [[internal chat](https://gitpod.slack.com/archives/C05H5UQBW6Q/p1709835458123889)]

## How to test

Check changes of test cases (there're some note for changes on the Description section)
- Run test in server
- Should have no error like `SubjectId mismatch` except what described on PR Description section

## Documentation


#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
